doc/SAI_LPO_Attributes.md

```diff
@@ -0,0 +1,141 @@
+
+### 2. Specification
+--------------------
+
+#### 2.1 Changes to saiport.h
+-----------------------------
+    /**
+     * @brief Serdes reach mode
+     */
+    typedef enum _sai_port_serdes_reach_mode_t
+    {
+        /** Normal Reach */
+        SAI_PORT_SERDES_REACH_MODE_NR,
+        /** Extended Reach */
+        SAI_PORT_SERDES_REACH_MODE_ER
+   
+    } sai_port_serdes_reach_mode_t;
+
+    /**
+     * @brief Serdes Rx Error Correcting Decoder/Maximum Likelihood
+     * Sequence Estimation control state
+     */
+    typedef enum _sai_port_serdes_rx_ecd_mlse_state_t
+    {
+        /** Disable */
+	SAI_PORT_SERDES_RX_ECD_MLSE_STATE_DISABLE,
+        /** Enable */
+        SAI_PORT_SERDES_RX_ECD_MLSE_STATE_ENABLE
+	
+    } sai_port_serdes_rx_ecd_mlse_state_t;
+
+    /**
+     * @brief Serdes polarity setting value
+     */
+    typedef enum _sai_port_serdes_polarity_t
+    {
+        /** Normal polarity */
+        SAI_PORT_SERDES_POLARITY_NORMAL,
+        /** Inverted polarity */
+        SAI_PORT_SERDES_POLARITY_INVERTED
+	
+    } sai_port_serdes_polarity_t;
+		
+
+The following attributes are added to `sai_port_serdes_attr_t`:
+
+    /**
+     * @brief Port serdes Tx upper eye non linear compensation percentage value
+     *
+     * List of port serdes Tx upper eye non linear compensation percentage value
+     * The values are of type sai_u32_list_t where the count is number of lanes
+     * in a port and the list specifies list of values to be applied to each
+     * lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_NLC_PERCENTAGE,
+
+    /**
+     * @brief Port serdes Tx lower eye non linear compensation percentage value
+     *
+     * List of port serdes Tx lower eye non linear compensation percentage value
+     * The values are of type sai_u32_list_t where the count is number of lanes
+     * in a port and the list specifies list of values to be applied to each
+     * lane.
+     *
+     * @type sai_u32_list_t
+     * @flags CREATE_AND_SET
+     * @default internal
+     */
+    SAI_PORT_SERDES_ATTR_TX_NLC_LOWER_EYE_PERCENTAGE,
+
+    /**
+     * @brief Port serdes reach mode control
+     *
+     * To select per port NR/ER mode for a port with back plane media type.
+     *
+     * @type sai_s32_list_t sai_port_serdes_reach_mode_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_PORT_SERDES_ATTR_REACH_MODE,
+
+    /**
+     * @brief Port serdes Rx Error Correcting Decoder/Maximum Likelihood
+     * Sequence Estimation control
+     *
+     * To enable/disable Rx ECD for a port with back plane media type.
+     *
+     * @type sai_s32_list_t sai_port_serdes_rx_ecd_mlse_state_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_PORT_SERDES_ATTR_RX_ECD_MLSE_STATE,

```

Comment:
Please also double-check if it is a standard attribute. See my comment above

Reply:
@eddyk-nvidia MLSE is a known technique similar to existing channel equalization techniques like DFE, FFE, CTLE. Though here its being introduced in the LPO context, its not a mandatory attribute as per the LPO MSA. Its upto the user if they want to derive the extra BER improved performance that comes with MLSE being enabled.